### PR TITLE
Clean up nuproj props / targets so that common behavior changes are m…

### PR DESCRIPTION
**Pull request by bbowman**
…ore easy to factor out.

### CODE REVIEW

File: build/WinObjC.Frameworks.UWP/WinObjC.Frameworks.UWP.Packaging.targets [hunk 1]
```diff
@@ -1,48 +1,6 @@
...
-       Properties="Configuration=$(Configuration);Platform=$(Platform)">
-      <Output TaskParameter="TargetOutputs" ItemName="_SolutionConfigurationContents" />
-    </MSBuild>
-
-    <Message Text="AddSingleConfigBuildOutput Platform: $(Platform)" Importance="High"/>
-
-    <MSBuild Projects="$(MSBuildProjectFile)"
-       Targets="GetPackageContents"
-       Properties="SkipGetPackageContentsForOtherPlatforms=true;BuildingInsideVisualStudio=false;BuildProjectReferences=true;Configuration=$(Configuration);Platform=$(Platform);CurrentSolutionConfigurationContents=%(_SolutionConfigurationContents.Identity);GetPackageContentsDependsOn=Build;$(GetPackageContentsDependsOn)">
-      <Output TaskParameter="TargetOutputs" ItemName="_PackageContentsForPlat" />
```
Comment (pradipd) [inline build/WinObjC.Frameworks.UWP/WinObjC.Frameworks.UWP.Packaging.targets]: I like all the red!

Comment (bbowman) [inline build/WinObjC.Frameworks.UWP/WinObjC.Frameworks.UWP.Packaging.targets]: @pradipd same!

File: tools/tools.sln [hunk 2]
```diff
@@ -278,16 +278,13 @@ Global
...
-		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|Any CPU.Build.0 = Release|Any CPU
-		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|ARM.ActiveCfg = Release|ARM
-		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|ARM.Build.0 = Release|ARM
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Debug|Any CPU.ActiveCfg = Debug|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Debug|ARM.ActiveCfg = Debug|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Debug|x86.ActiveCfg = Debug|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Debug|x86.Build.0 = Debug|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|Any CPU.ActiveCfg = Release|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|Any CPU.Build.0 = Release|x86
+		{0A23B6B4-94BB-4B3A-BA73-A7CB5EF7D8F9}.Release|ARM.ActiveCfg = Release|x86
```
Comment (memontic-ms) [inline tools/tools.sln]: Why did this change?

File: common/winobjc.nuproj.common.targets [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    <GetPackageVersionDependsOn>SetPackageVersion;$(GetPackageVersionDependsOn);</GetPackageVersionDependsOn>
+  </PropertyGroup>
+
+  <Target Name="SetPackageVersion" DependsOnTargets="GetVersion" >
+    <PropertyGroup>
+        <PackageVersion></PackageVersion>
+    </PropertyGroup>
+
+    <Error Condition="'$(GitVersion_BranchName)' == ''"
+           Text="Cannot determine git branch. Please make sure GitInfo is listed in the project.json for this project and that git.exe is part of the PATH environment variable." />
```
Comment (memontic-ms) [inline common/winobjc.nuproj.common.targets]: >GitInfo [](start = 63, length = 7)

Change to GitVersionTask

File: common/winobjc.nuproj.common.targets [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      <PackageVersion_Major>$(GitVersion_Major)</PackageVersion_Major>
+      <PackageVersion_Minor>$(GitVersion_Minor)</PackageVersion_Minor>
+      
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' == ''">$(GitVersion_Patch)</PackageVersion_Patch>
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' != ''">$([MSBuild]::Add('$(GitVersion_Patch)', '1'))</PackageVersion_Patch>
+
+      <PackageVersion_PreReleaseLabel>$(GitVersion_PreReleaseLabel)</PackageVersion_PreReleaseLabel>
+      <PackageVersion_Timestamp>$(PackageTimestamp)</PackageVersion_Timestamp>
+
+      <!-- Replace PreReleaseFormat msbuild propert names with positional String.Format args.
```
Comment (memontic-ms) [inline common/winobjc.nuproj.common.targets]: >propert [](start = 44, length = 7)

nit: property

File: common/winobjc.nuproj.common.targets [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+           NOTE: If additional porperty replacements are needed, they must show up here.
```
Comment (memontic-ms) [inline common/winobjc.nuproj.common.targets]: >porperty [](start = 31, length = 8)

nit: you really don't like this word

File: build/CFNetwork/dll/CFNetwork.vcxproj [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(SolutionDir)\..\common\winobjc.nuproj.common.props" Condition="Exists('$(SolutionDir)\..\common\winobjc.nuproj.common.props')" />
```
Comment (memontic-ms) [inline build/CFNetwork/dll/CFNetwork.vcxproj]: >$(SolutionDir)\..\common\winobjc.nuproj.common.props [](start = 92, length = 52)

This path seems wrong

Comment (memontic-ms) [inline build/CFNetwork/dll/CFNetwork.vcxproj]: Why is this import even needed for CFNetwork, CoreFoundation, Foundation, MobileCoreServices, Starboard..

Comment (memontic-ms) [inline build/CFNetwork/dll/CFNetwork.vcxproj]: Nevermind

---
In reply to: [105063164](https://github.com/Microsoft/WinObjC/pull/2189#discussion_r105063164) [](ancestors = 105063164)

File: common/winobjc.nuproj.common.props [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <OutputPath>$(SolutionDir)\OutputPackages\$(Configuration)\</OutputPath>
+    <InferLegacyPackageReferences>false</InferLegacyPackageReferences>
+    <IncludeSymbolsInPackage>false</IncludeSymbolsInPackage>
+    <IncludeFrameworkReferencesInPackage>false</IncludeFrameworkReferencesInPackage>
+    <InferLegacyPackageReferences>false</InferLegacyPackageReferences>
```
Comment (memontic-ms) [inline common/winobjc.nuproj.common.props]: ><InferLegacyPackageReferences>false</InferLegacyPackageReferences> [](start = 4, length = 66)

Duplicate of line 6

**Pull request merged**